### ISSUE TITLE
Unexpected failure when preparing tensor allocations: tensorflow/lite/kernels/transpose.cc:

### ISSUE THREAD

**Issue by umair13adil** [tensorflow/tensorflow#40031]
I have trained an audio classification model using Keras & Tensorflow and converted it to Tensorflow lite, it is converted fine but when I run on Android I get this error:

```
 Process: org.tensorflow.lite.examples.speech, PID: 15170
    java.lang.IllegalStateException: Internal error: Unexpected failure when preparing tensor allocations: tensorflow/lite/kernels/transpose.cc:56 op_context->perm->dims->data[0] != dims (3 != 2)
    Node number 1 (TRANSPOSE) failed to prepare.
    
        at org.tensorflow.lite.NativeInterpreterWrapper.allocateTensors(Native Method)
        at org.tensorflow.lite.NativeInterpreterWrapper.run(NativeInterpreterWrapper.java:145)
        at org.tensorflow.lite.Interpreter.runForMultipleInputsOutputs(Interpreter.java:314)
```

**Usage:**

```
tfLite = try {
    tfliteOptions.setNumThreads(4)

    Interpreter(loadModelFile(assets, actualModelFilename), tfliteOptions)
} catch (e: Exception) {
    throw RuntimeException(e)
}

for (i in 0 until RECORDING_LENGTH) {
    floatInputBuffer[i][0] = inputBuffer[i] / 32767.0f
}

val inputArray = arrayOf<Any>(floatInputBuffer)
val outputMap: MutableMap<Int, Any> = HashMap()
outputMap[0] = outputScores

// Run the model.
tfLite?.runForMultipleInputsOutputs(inputArray, outputMap)
```

**Android Dependency:**

` implementation 'org.tensorflow:tensorflow-lite:2.2.0'`
       
**Python Requirements:**

```
tensorboard==2.1.1
tensorflow==2.1.0
tensorflow-estimator==2.1.0
```

**Model Conversion:**

    # Create a converter
    converter = tf.lite.TFLiteConverter.from_keras_model(model)
    converter.allow_custom_ops = True
    converter.experimental_new_converter = True

    # Convert the model
    tflite_model = converter.convert()

    # Create the tflite model file
    tflite_model_name = "android/app/src/main/assets/converted_model.tflite"
    open(tflite_model_name, "wb").write(tflite_model)

**Model Summary:**

Model: "long_short_term_memory"

|Layer (type)|Output Shape|Param #| Connected to|

|input (InputLayer) |             [(None, 1, 16000)]|   0    |                                        
melbands (Melspectrogram)       (None, 128, 100, 1)  296064      input[0][0]                      
batch_norm (Normalization2D)    (None, 128, 100, 1)  0           melbands[0][0]                   
permute (Permute)               (None, 100, 128, 1)  0           batch_norm[0][0]                 
reshape (TimeDistributed)       (None, 100, 128)     0           permute[0][0]                    
d_dense_tanh (TimeDistributed) (None, 100, 64)      8256        reshape[0][0]                    
bidirectional_lstm (Bidirection (None, 100, 64)      24832       td_dense_tanh[0][0]              
skip_connection (Concatenate)   (None, 100, 128)     0           td_dense_tanh[0][0]              
                                                                 bidirectional_lstm[0][0]         
dense_1_relu (Dense)            (None, 100, 64)      8256        skip_connection[0][0]            
max_pool_1d (MaxPooling1D)      (None, 50, 64)       0           dense_1_relu[0][0]               
dense_2_relu (Dense)            (None, 50, 32)       2080        max_pool_1d[0][0]                
flatten (Flatten)               (None, 1600)         0           dense_2_relu[0][0]               
dropout (Dropout)               (None, 1600)         0           flatten[0][0]                    
dense_3_relu (Dense)            (None, 32)           51232       dropout[0][0]                    
softmax (Dense)                 (None, 2)            66          dense_3_relu[0][0]               

Total params: 390,786
Trainable params: 390,786
Non-trainable params: 0

**Complete Python & Android Code can be found here:**
[TensorFlow Lite Audio Classification](https://github.com/umair13adil/Audio-Classification)

**Comment by Saduf2019** [comment]
@umair13adil 
I ran the code shared by you there is syntax error, can you please share simple indented stand alone code such that we can replicate the issue or if possible share a colab gist to analyse the issue.

Can you please refer to issues with similar error message: #36555 #38926 if it helps.

**Comment by umair13adil** [comment]
@Saduf2019 The Issue is on the Android side. Tensorflow lite library is crashing when I run it, If you run the Android code in Android studio, you will see the exception details. 

Can you tell me what I can do to optimize my model so Tensorflow lite can run without crashing?

**Comment by umair13adil** [comment]
I just updated these dependencies in python:

    tensorboard==2.2.2
    tensorflow==2.2.0
    tensorflow-estimator==2.2.0

After I built my model, it started working on Android. So the issue was related to the incompatible version on the Python side.

**Comment by PatroxGaurab** [comment]
I'm facing the same issue on the latest version of tensorflow
Can you guide me how to debug it?

**Comment by jvishnuvardhan** [comment]
@PatroxGaurab This issue is old and already got closed. Can you please create a new issue with simple standalone code to reproduce the issue? Thanks!
